libsolidity/analysis/TypeChecker.cpp

```diff
@@ -336,12 +340,16 @@ bool TypeChecker::visit(FunctionDefinition const& _function)
 		else if (_function.libraryFunction())
 			m_errorReporter.typeError(7801_error, _function.location(), "Library functions cannot be \"virtual\".");
 	}
+	if (_function.overrides() && _function.isFree())
+		m_errorReporter.syntaxError(1750_error, _function.location(), "Free functions cannot override.");
 
 	if (_function.isPayable())
 	{
 		if (_function.libraryFunction())
 			m_errorReporter.typeError(7708_error, _function.location(), "Library functions cannot be payable.");
-		if (_function.isOrdinary() && !_function.isPartOfExternalInterface())
+		else if (_function.isFree())
+			m_errorReporter.typeError(9559_error, _function.location(), "Free functions cannot be payable.");
```

Comment:
If we are placing so many restrictions, I think the "free functions" term is pretty much a misnomer, and based on the documentation piece, these would be just called "inline functions".

Reply:
internal?

Reply:
I guess they are kind of library functions but free from libraries